### ISSUE TITLE
Decorator to exclude functions from runtime type checks

### ISSUE THREAD

**Issue by cjolowicz** [agronholm/typeguard#179]
Would you accept a PR adding a decorator that excludes functions from runtime type checks with Typeguard, while still allowing for static type checks?

**Is your feature request related to a problem? Please describe.**

My primary use of Typeguard is as a mandatory check in CI using the pytest plugin. Typeguard has repeatedly identified bugs that were invisible to static type checkers. Unfortunately, this also means that I regularly run into problems due to discrepancies between static and runtime type-checking. Here are some examples from Typeguard's issue tracker:

- #178
- #138
- #115

We briefly talked about this before, in #138.

**Describe the solution you'd like**

Here's a POC:

```python
from typing import TypeVar
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    F = TypeVar("F")
    def typeguard_ignore(f: F) -> F:
        return f
else:
    from typing import no_type_check as typeguard_ignore

@typeguard_ignore
def f(arg: int) -> int:
    return arg

def test_f() -> None:
    f("teapot")
```

This code passes at runtime:

```sh
$ pytest --typeguard-packages=repro repro.py
...
1 passed in 0.01s
```

...but it's still caught by static type checkers:

```sh
$ mypy repro.py
repro.py:19: error: Argument 1 to "f" has incompatible type "str"; expected "int"
```

**Describe alternatives you've considered**

- Defining `typeguard_ignore` downstream (quite happy to do this if you prefer)
- Using `@no_type_check` (loses static type checks)
- Finding a work around for the special case (not always possible)

**Comment by agronholm** [comment]
I would accept a PR, but make sure it works with the import hook too.
Discrepancies between the two are a reality, plus as you pointed out, typeguard has several outstanding issues regarding how it does the checks. I plan to address most of these in v3.0 whenever I get around to that (my focus is on a couple other projects now) so this would do nicely as an interim solution.

**Comment by MVrachev** [comment]
When a release is planned? 
Wanted to use that feature, but is not yet available.

**Comment by agronholm** [comment]
I'll make a release this week.

**Comment by agronholm** [comment]
A little bit late but I've now released v2.12.0.

**Comment by cjolowicz** [comment]
Thank you @agronholm :)
